docs/articles/en/advanced-usage.md

```diff
@@ -0,0 +1,356 @@
+
+## Script Types and Execution Modes
+
+### Local File Execution
+
+Execute C# files with custom entry points:
+
+```sh
+# Default entry point (Main method)
+dotnet-exec MyScript.cs
+
+# Custom entry point
+dotnet-exec MyScript.cs --entry MainTest
+
+# Multiple entry methods (fallback)
+dotnet-exec MyScript.cs --default-entry MainTest Execute Run
+```
+
+### Remote File Execution
+
+Execute scripts directly from URLs:
+
+```sh
+# GitHub raw file
+dotnet-exec https://raw.githubusercontent.com/user/repo/main/script.cs
+
+# Any accessible URL
+dotnet-exec https://example.com/scripts/sample.cs
+```
+
+### Multiple Script Files
+
+Execute multiple scripts together:
+
+```sh
+# Multiple files as arguments
+dotnet-exec Script1.cs Script2.cs Script3.cs
+
+# Using additional scripts option
+dotnet-exec MainScript.cs --addition Helper1.cs --addition Helper2.cs
+```
+
+## Advanced Reference Management
+
+### Framework References
+
+Target specific framework features:
+
+```sh
+# Web framework (ASP.NET Core)
+dotnet-exec 'WebApplication.Create().Run();' --web
+
+# Explicit framework reference
+dotnet-exec 'WebApplication.Create().Run();' --reference 'framework:web'
+
+# Multiple framework references
+dotnet-exec MyScript.cs --reference 'framework:web' --reference 'framework:desktop'
+```
+
+### Project References
+
+Use existing project dependencies:
+
+```sh
+# Reference a project file
+dotnet-exec MyScript.cs --reference 'project:./MyLibrary.csproj'
+
+# Extract all references from project
+dotnet-exec MyScript.cs --project ./MyProject.csproj
+```
+
+### Complex NuGet References
+
+```sh
+# Specific version
+dotnet-exec MyScript.cs -r 'nuget:Newtonsoft.Json,13.0.3'
+
+# Prerelease versions
+dotnet-exec MyScript.cs -r 'nuget:Microsoft.Extensions.Hosting,8.0.0-preview.1'
+
+# Multiple packages
+dotnet-exec MyScript.cs \
+  -r 'nuget:Serilog' \
+  -r 'nuget:Serilog.Sinks.Console' \
+  -u 'Serilog'
+```
+
+## Advanced Using Statements
+
+### Static Imports
+
+```sh
+# Static using for Console methods
+dotnet-exec 'WriteLine("Hello World");' --using 'static System.Console'
+
+# Multiple static imports
+dotnet-exec MyScript.cs \
+  --using 'static System.Console' \
+  --using 'static System.Math'
+```
+
+### Using Aliases
+
+```sh
+# Create type aliases
+dotnet-exec 'Json.SerializeObject(data);' \
+  --using 'Json = Newtonsoft.Json.JsonConvert' \
+  -r 'nuget:Newtonsoft.Json'
+
+# Namespace aliases
+dotnet-exec MyScript.cs \
+  --using 'Extensions = Microsoft.Extensions.DependencyInjection'
+```
+
+### Removing Default Usings
+
+```sh
+# Remove default System namespace
+dotnet-exec 'System.Console.WriteLine("Hello");' --using '-System'
+
+# Remove multiple namespaces
+dotnet-exec MyScript.cs --using '-System' --using '-System.Collections.Generic'
+```
+
+## Compilation and Execution Options
+
+### Preview Features
+
+```sh
+# Enable preview language features
+dotnet-exec PreviewScript.cs --preview
+
+# With specific preview APIs
```

Comment:
Enabled features that require `RequiresPreviewFeaturesAttribute` or C# features in preview